src/coreclr/jit/lowerxarch.cpp

```diff
@@ -5508,20 +5596,55 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* containingNode, Ge
         }
     }
 
-    noway_assert(supportsRegOptional != nullptr);
     *supportsRegOptional = supportsGeneralLoads;
 
     if (!node->OperIsHWIntrinsic())
     {
-        return supportsGeneralLoads && IsContainableMemoryOp(node);
+        return supportsGeneralLoads && (IsContainableMemoryOp(node) || node->IsCnsNonZeroFltOrDbl());
```

Comment:
Why does this extend support to __non-zero__ floating point constants and not to all floating point constants?
Is the expectation here that `Vector128.CreateScalarUnsafe(0.0)` is going to be replaced with `Vector128<double>.Zero`?

Reply:
Yes, its because `Zero` nodes shouldn't come from memory since its more efficient to just emit `xorps`. `CreateScalarUnsafe(0)` should just generate `xorps` since the upper bits don't matter and that's the most efficient way of getting zero in the first element. 

@TIHan should be able to cover this scenario in the WIP he's already working on since its directly related: https://github.com/dotnet/runtime/pull/63821

Reply:
Okay, thanks for confirming.